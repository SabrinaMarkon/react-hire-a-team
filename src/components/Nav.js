import React from 'react';
import { Route, NavLink, BrowserRouter as Router, Switch } from 'react-router-dom';
import App from '../containers/App';
import HowItWorks from '../components/HowItWorks';
import Signup from '../components/Signup';
import Login from '../components/Login';
import PostJob from '../components/PostJob';

export default function Nav() {

  return (
    <Router>
      <nav>
          <ul>
            <li>
              <NavLink exact activeClassName="active" to="/">Home</NavLink>
            </li>
            <li>
              <NavLink activeClassName="active" to="/howitworks">How It Works</NavLink>
            </li>
            <li>
              <NavLink activeClassName="active" to="/signup">Signup</NavLink>
            </li>
            <li>
              <NavLink activeClassName="active" to="/login">Login</NavLink>
            </li>
            <li>
              <NavLink activeClassName="active" to="/postjob">Post a Job</NavLink>
            </li>
          </ul>

      </nav>
    </Router>
  )
}