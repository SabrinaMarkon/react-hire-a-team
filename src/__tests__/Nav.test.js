import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

// components
import Nav from '../components/Nav';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Nav />);
});

describe('Testing Nav component', () => {
  it('renders the navigation', () => {
    const nav = document.createElement('nav');
    ReactDOM.render(<Nav />, nav);
    ReactDOM.unmountComponentAtNode(nav);
  });

  it('should render 1 child Material-UI AppBar component', () => {
    expect(wrapper.find('AppBar')).toHaveLength(1);
  });

  it('it should match the Nav snapshot', () => {
    // If snapshot exists, compare. If not, create the snapshot.
    expect(wrapper).toMatchSnapshot();
  });
});