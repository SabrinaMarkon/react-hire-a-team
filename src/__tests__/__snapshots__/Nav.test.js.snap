// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Nav component it should match the Nav snapshot 1`] = `
<nav>
  <WithStyles(ForwardRef(AppBar))
    color="primary"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        variant="h1"
      >
        Hire-A-Team
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</nav>
`;
